### PR TITLE
Add parameters to UVCal and calfits to support sky-based calibration

### LINKED ISSUE #289: calfits flexibility for sky-based cal
We need to incorporate some header keyword flexibility that will help as we start including sky-based calibrations (e.g. FHD). Here are some suggestions.

New Required keyword: 
cal_style. Two options I can think of would be (redundant, sky).
Required keywords given the cal_style : 
sky_field (required if cal_style is sky): short string describing field center or dominant source.
sky_catalog (required if cal_style is sky): catalog that was used
ref_antenna_name (required if cal_style is sky): name of antenna that was used as the phase reference antenna
Optional keywords:
n_sources (optional if cal_style is sky): number of sources included in the calibration model
cal_baseline_range (optional if cal_style is sky): range in wavelengths of baselines included in calibration.
diffuse_model (optional if cal_style is sky): short string describing calibration diffuse model used.

### PR DESCRIPTION
**Pull request by bhazelton**
This fixes #289

### CODE REVIEW

File: pyuvdata/calfits.py [hunk 1]
```diff
@@ -109,11 +129,6 @@ def write_calfits(self, filename, run_check=True, check_extra=True,
             prihdr['FRQRANGE'] = ','.join(map(str, self.freq_range))
         prihdr['TMERANGE'] = ','.join(map(str, self.time_range))
 
-        for p in self.extra():
```
Comment (plaplant) [inline pyuvdata/calfits.py]: So does this mean that we're not supporting arbitrary extra parameters anymore? I think that's okay, just wondering if this was added somewhere else and I missed it.

File: pyuvdata/calfits.py [hunk 2]
```diff
@@ -413,7 +440,7 @@ def read_calfits(self, filename, run_check=True, check_extra=True,
             self.Nspws = hdr.pop('NAXIS5')
             # add this for backwards compatibility when the spw CRVAL wasn't recorded
             try:
-                spw_array = uvutils.fits_gethduaxis(F[0], 5, strict_fits=strict_fits) - 1
+                spw_array = uvutils.fits_gethduaxis(F[0], 5, strict_fits=strict_fits)
```
Comment (plaplant) [inline pyuvdata/calfits.py]: Good catch!

File: pyuvdata/uvcal.py [hunk 3]
```diff
@@ -190,6 +196,41 @@ def __init__(self):
                                            description=desc, form=(2,),
                                            expected_type=float, tols=1e-3)
 
+        # --- cal_style parameters ---
+        desc = ('Style of calibration. Values are sky or redundant.')
```
Comment (plaplant) [inline pyuvdata/uvcal.py]: This is already defined above. Should it be defined again here?

File: pyuvdata/uvcal.py [hunk 4]
```diff
@@ -250,6 +291,11 @@ def check(self, check_extra=True, run_check_acceptability=True):
             run_check_acceptability: Option to check if values in required parameters
                 are acceptable. Default is True.
         """
+        # if make sure requirements are set properly for cal_style
```
Comment (plaplant) [inline pyuvdata/uvcal.py]: Small typo in the comment

File: pyuvdata/calfits.py (hunk 1 above)
Comment (bhazelton) [reply pyuvdata/calfits.py]: We are, but they go in extra_keywords. This part of the code was apparently to catch actual parameters (meaning ones that are defined explicitly on the object) that were not required to be set and that we didn't want to bother adding to the header explicitly. The problem is that if they need any special handling (like baseline_range) then you have to do them explicitly and then they get caught here as well, which is bad. Also, the keyword in the FITS file gets set by simple truncation which tends to give them somewhat confusing or unclear names and can lead to redundant names. I think this was a bit lazy and caused lots of issues.

File: pyuvdata/calfits.py (hunk 2 above)
Comment (bhazelton) [reply pyuvdata/calfits.py]: Yeah, I'm not sure how that made it through before, but I had a bug that I was hunting and I found that one (it didn't fix my bug, but I found it somewhere else).

File: pyuvdata/uvcal.py (hunk 3 above)
Comment (bhazelton) [reply pyuvdata/uvcal.py]: oops! thank you! I'll remove one of them

File: pyuvdata/uvcal.py (hunk 4 above)
Comment (bhazelton) [reply pyuvdata/uvcal.py]: thanks!

File: pyuvdata/uvcal.py (hunk 3 above)
Comment (bhazelton) [reply pyuvdata/uvcal.py]: I took out the upper one.

**Pull request merged**